climada/engine/impact.py

```diff
@@ -84,164 +487,157 @@ class Impact():
         only filled if save_mat is True in calc()
     """
 
-    def __init__(self):
-        """Empty initialization."""
-        self.tag = dict()
-        self.event_id = np.array([], int)
-        self.event_name = list()
-        self.date = np.array([], int)
-        self.coord_exp = np.ndarray([], float)
-        self.crs = DEF_CRS
-        self.eai_exp = np.array([])
-        self.at_event = np.array([])
-        self.frequency = np.array([])
-        self.tot_value = 0
-        self.aai_agg = 0
-        self.unit = ''
-        self.imp_mat = sparse.csr_matrix(np.empty((0, 0)))
+    def __init__(self,
+                 event_id=None,
+                 event_name=None,
+                 date=None,
+                 frequency=None,
+                 coord_exp=None,
+                 crs=DEF_CRS,
+                 eai_exp=None,
+                 at_event=None,
+                 tot_value=0,
+                 aai_agg=0,
+                 unit='',
+                 imp_mat=None,
+                 tag=None):
+
+        self.tag = tag or {}
+        self.event_id = np.array([], int) if event_id is None else event_id
+        self.event_name = event_name or []
+        self.date = np.array([], int) if date is None else date
+        self.coord_exp = np.ndarray([], float) if coord_exp is None else coord_exp
+        self.crs = crs
+        self.eai_exp = np.array([], float) if eai_exp is None else eai_exp
+        self.at_event = np.array([], float) if at_event is None else at_event
+        self.frequency = np.array([],float) if frequency is None else frequency
+        self.tot_value = tot_value
+        self.aai_agg = aai_agg
+        self.unit = unit
+        self.imp_mat = sparse.csr_matrix(np.empty((0, 0))) if imp_mat is None else imp_mat
```

Comment:
Same as for `ImpactCalc`:
```suggestion
        self.imp_mat = imp_mat
```